File: NoAppNoGain/app/src/main/java/com/example/noappnogain/Reinitialiser.java
package com.example.noappnogain;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.FirebaseAuth;

public class Reinitialiser extends AppCompatActivity {
    private EditText passwordEmail;
    private Button resetpassword;
    private FirebaseAuth firebaseAuth;

    public Reinitialiser() {
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_reinitialiser_mdp);

        passwordEmail = (EditText)findViewById(R.id.pass_email);
        resetpassword = (Button)findViewById(R.id.reset_pass);
        firebaseAuth = FirebaseAuth.getInstance();

        resetpassword.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String useremail = passwordEmail.getText().toString().trim();

                if(useremail.equals(""))
                {
                    passwordEmail.setError("E-mail obligatoire...",null);
                    return;
                }
                else
                {
                    firebaseAuth.sendPasswordResetEmail(useremail).addOnCompleteListener(new OnCompleteListener<Void>() {
                        @Override
                        public void onComplete(@NonNull Task<Void> task) {
                            if(task.isSuccessful())
                            {
                                Toast.makeText(Reinitialiser.this,"E-mail envoyé avec succès..",Toast.LENGTH_LONG).show();
                                finish();
                                startActivity(new Intent(Reinitialiser.this,Sidentifier.class));
                            }
                            else
                            {
                                Toast.makeText(Reinitialiser.this,"Erreur lors de l'envoi de l'e-mail..",Toast.LENGTH_SHORT).show();
                            }
                        }
                    });
                }
            }
        });
    }
}

File: NoAppNoGain/app/src/main/java/com/example/noappnogain/Reinitialiser.kt
package com.example.noappnogain

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth

class Reinitialiser : AppCompatActivity() {
    private lateinit var passwordEmail: EditText
    private lateinit var resetpassword: Button
    private lateinit var firebaseAuth: FirebaseAuth
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_reinitialiser_mdp)
        passwordEmail = findViewById<View>(R.id.pass_email) as EditText
        resetpassword = findViewById<View>(R.id.reset_pass) as Button
        firebaseAuth = FirebaseAuth.getInstance()
        resetpassword.setOnClickListener(View.OnClickListener {
            val useremail = passwordEmail.text.toString().trim { it <= ' ' }
            if (useremail == "") {
                passwordEmail.setError("E-mail obligatoire...", null)
                return@OnClickListener
            } else {
                firebaseAuth.sendPasswordResetEmail(useremail).addOnCompleteListener { task ->
                    if (task.isSuccessful) {
                        Toast.makeText(
                            this@Reinitialiser,
                            "E-mail envoyé avec succès..",
                            Toast.LENGTH_LONG
                        ).show()
                        finish()
                        startActivity(Intent(this@Reinitialiser, Sidentifier::class.java))
                    } else {
                        Toast.makeText(
                            this@Reinitialiser,
                            "Erreur lors de l'envoi de l'e-mail..",
                            Toast.LENGTH_SHORT
                        ).show()
                    }
                }
            }
        })
    }
}